plugins/chat/app/services/chat/create_thread.rb

```diff
@@ -30,50 +30,47 @@ class CreateThread
     policy :threading_enabled_for_channel
     model :original_message
     transaction do
-      step :find_or_create_thread
+      model :thread, :find_or_create_thread
```

Comment:
This will automatically assign `context[:thread]` and check if the resulting model is valid or not.